### PR TITLE
Add option to build with O3 level

### PR DESCRIPTION
**Pull request by skvl**
Relates to https://github.com/libvmi/libvmi/pull/1059

### CODE REVIEW

**Comment by tklengyel** [comment]
O3 should not be separate from HARDENING. The two modes should be compatible, so even with O3 enabled you should keep the hardening options unless hardening is explicitely disabled.

**Comment by tklengyel** [comment]
Also, please add the equivalent option to the meson system, I'm hoping to transition completely over to meson as the default this year while keeping autoconf as a backup.

File: meson.build [hunk 1]
```diff
@@ -55,6 +55,14 @@ add_project_arguments('-Wno-reorder-init-list', language : ['cpp'])
 add_project_arguments('-Wno-unknown-warning-option', language : ['cpp'])
 add_project_arguments('-Wno-unused-parameter', language : ['c', 'cpp'])
 
+if get_option('buildtype') == 'release'
+  add_project_arguments('-O3',  language : ['c', 'cpp'])
```
Comment (tklengyel) [inline meson.build:59]: If O3 is incompatible with hardening, shouldn't this be in the `else` clause of checking hardening is enabled (which is by default)?

File: meson.build [hunk 2]
```diff
@@ -55,6 +55,14 @@ add_project_arguments('-Wno-reorder-init-list', language : ['cpp'])
...
+endif
+
+if get_option('buildtype') == 'debugoptimized'
+  add_project_arguments('-O3', '-g3',  language : ['c', 'cpp'])
+endif
```
Comment (eli-schwartz) [inline meson.build:64]: Actually I would say that this is all wrong. The builtin "buildtype" option is specifically supposed to handle this for you:
https://github.com/xmake-io/xmake-repo/pull/1837#issuecomment-1432510233

For release it already adds -O3.
For debugoptimized it already adds `-g2 -O2`, which is usually considered a pretty decent balance...

**Comment by tklengyel** [comment]
Closing PR for now due to inactivity.

**Pull request closed without merging**